src/TraceEvent/Stacks/SpeedScopeExporter.cs

```diff
@@ -0,0 +1,260 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Globalization;
+using System.IO;
+using System.Linq;
+
+namespace Microsoft.Diagnostics.Tracing.Stacks
+{
+    public static class SpeedScopeExporter
+    {
+        internal struct Sample
```

Comment:
I would like code to follow the following convention. 
1. Public APIs some first, and generally in an order that makes sense from a consumers' point of view (e.g. constructors first, most important/popular APIs first ...)
2. All public APIs have C# XML doc comments, at least the 'summary' part.  You don't need to do the arguments / return elements of doc comments if the summary sufficiently explains them (you will see that mostly the doc comments only have the summary part).
3. All private/internal things are in a #region private , so that they are collapsed aggressively by VS.  Thus when you look at the API you only see the public stuff (and their comments), which is what you want)
4. When possible (the fields are private)  state (fields) come last in the class (typically in the private region.   This lets you see what is 'in' a class quickly.  
5. Types that are private to one class are nested in that class (in the private region.

Can you rearrange your code to this standard?   Thanks

Reply:
> Can you rearrange your code to this standard?

Done